Feature/43 crud table price

stishchenko opened the pull request.

stishchenko committed:
#43 CRUD Price table
Price -> create POJO class.
PriceState -> create enum.
PriceService -> create service for crud.
PriceDao -> create interface to connect to db.
PriceDaoImpl -> temp implementation of PriceDao.

stishchenko committed:
#43 validation and test
PriceService -> add validation.
PriceDao -> change param names.
PriceDaoTestImpl -> test implementation of PriceDao.
PriceDataTest -> price for test.
PriceServiceTest -> create to test PriceService.

stishchenko committed:
#43 validation and test
PriceService -> refactor createPrice() validation.
PriceServiceTest -> fill tests for PriceService.

stishchenko committed:
#43
PriceDao -> rename methods. In other files they were refactored too.

rdovgan commented on src/main/java/team/asd/dao/PriceDaoImpl.java:
```diff
@@ -0,0 +1,48 @@
+		return id;
+	}
+
+	private Price getExpectedPrice() {
```
@stishchenko this method should not be here

rdovgan commented on src/main/resources/team/asd/mapper/ProductMapper.xml:
```diff
@@ -0,0 +1,57 @@
+    </resultMap>
+
+    <select id="readProductById" parameterType="java.lang.Integer" resultMap="productMap">
+        select *
```
@stishchenko please do not use asterisks for `select` queries

stishchenko committed:
Merge branch 'feature/39_DB_layer_for_ProductService' into feature/43_CRUD_table_Price

stishchenko committed:
#43
Price* -> change return types on void, refactor tests.
ProductMapper.xml -> replace asterisk with column names.

stishchenko committed:
Merge branch 'master' into feature/43_CRUD_table_Price

rdovgan reviewed (approved):
👍🏻
